### PR TITLE
Restart kube-vip pods after updating ingress-expose svc

### PR DESCRIPTION
**Pull request by yaocw2020**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

The VIP is temporarily not accessible during upgrading. 

**Solution:**

The Harvester pod adds the annotation `kube-vip.io/ignore-service-security: "true"` to the service ingress-expose, but the kube-vip can't handle the modification. Just delete the kube-vip pods to let the kube-vip watches service and handle the update even.

**Related Issue:** https://github.com/harvester/harvester/issues/4221

**Test plan:**

- Spin up a Harvester environment with version v1.2.0-rc5
- Delete annotation `kube-vip.io/ignore-service-security: "true"` of the service `kube-system/ingress-expose`.
- Delete kube-vip pod, and we will be unable to ping the VIP.
- Replace the Harvester image in the Harvester mangedchart. We'll be able to ping the VIP after the Harvester updates the service and restart the kube-vip.

### CODE REVIEW

File: pkg/controller/master/rancher/register.go [hunk 1]
```diff
@@ -110,75 +114,90 @@ func (h *Handler) registerExposeService() error {
...
+	svcCopy := svc.DeepCopy()
+	if svc.Annotations == nil {
+		svcCopy.Annotations = make(map[string]string)
+	}
+	svcCopy.Annotations[keyKubevipIgnoreServiceSecurity] = trueStr
+	if _, ok := svcCopy.Annotations[keyKubevipLoadBalancerIPs]; !ok {
+		svcCopy.Annotations[keyKubevipLoadBalancerIPs] = svc.Spec.LoadBalancerIP
+	}
+	if _, err := h.Services.Update(svcCopy); err != nil {
+		return err
```
Comment (w13915984028) [inline pkg/controller/master/rancher/register.go:197]: when `Services.Update` is successful but `restartKubevipPod` is failed, seems `restartKubevipPod` has no chance to be rescheduled

Comment (yaocw2020) [reply pkg/controller/master/rancher/register.go:197]: Yes, considering the time limit, the PR is just a workaround. I'll submit a PR to kube-vip to address the problem thoroughly.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks for the fix, also @yaocw2020 can u please help to file an new to track the upstream VIP enhancement of this, thanks.

**Review by bk201 — APPROVED** [review]
Test upgrade and VIP is alive.

**Review by Vicente-Cheng — APPROVED** [review]
LGTM, tested with this PR. The VIP looks fine.

**Comment by yaocw2020** [comment]
> LGTM, thanks for the fix, also @yaocw2020 can u please help to file an new to track the upstream VIP enhancement of this, thanks.

https://github.com/harvester/harvester/issues/4407

**Review by w13915984028 — APPROVED** [review]
LGTM, thanks.

**Comment by guangbochen** [comment]
@mergify backport v1.2

**Pull request merged**